pkg/server/container_execsync.go

```diff
@@ -115,14 +116,17 @@ func (c *criContainerdService) ExecSync(ctx context.Context, r *runtime.ExecSync
 			TypeUrl: runtimespec.Version,
 			Value:   rawSpec,
 		},
+		// ExecID is not used now. Containerd client will use it.
+		ExecID: execID,
 	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to exec in container %q: %v", id, err)
 	}
-	exitCode, err := waitContainerExec(cancel, events, id, resp.Pid, r.GetTimeout())
+	exitCode, err := c.waitContainerExec(ctx, eventstream, id, execID, r.GetTimeout())
```

Comment:
In fact, for exec we could use either pid or execID. The execID is introduced in containerd just to avoid a race. I changed the code to use execID just to be align with containerd.